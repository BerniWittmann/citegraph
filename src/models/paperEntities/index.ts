import Author from './author'
import Conference from './conference'
import Country from './country'
import Institution from './institution'
import Publication from './publication'
import Publisher from './publisher'
import Record from './record'
import PaperEntity, { PaperEntityTableColumn, PaperEntityFields } from './base'

export const entities = [
  Record,
  Author,
  Institution,
  Country,
  Publication,
  Publisher,
  Conference
]

export const entityKeysMap = {
  [Record.key]: Record,
  [Author.key]: Author,
  [Institution.key]: Institution,
  [Country.key]: Country,
  [Publication.key]: Publication,
  [Publisher.key]: Publisher,
  [Conference.key]: Conference
}

export const entityKeys: Array<string> = entities.map((entity) => entity.key)

export type PaperEntityTableColumn = PaperEntityTableColumn

export type PaperEntity = PaperEntity

export type PaperEntityFields = PaperEntityFields

export default {
  Author,
  Conference,
  Country,
  Institution,
  Publication,
  Publisher,
  Record
}
