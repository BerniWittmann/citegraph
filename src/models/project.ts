export default class Project {
  name: string;
  id?: number;

  constructor ({ name, id }: Project) {
    this.id = id
    this.name = name
  }
}
