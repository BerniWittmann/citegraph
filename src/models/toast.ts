import uuid from 'uuid'

export enum ToastSeverity {
  SUCCESS = 'success',
  INFO = 'info',
  WARNING = 'warning',
  ERROR = 'error'
}

export enum ToastIcon {
  SUCCESS = 'mdi-check-circle-outline',
  INFO = 'mdi-information-outline',
  WARNING = 'mdi-alert',
  ERROR = 'mdi-alert-circle-outline',
}

export enum ToastDelay {
  SUCCESS = 3000,
  INFO = 4000,
  WARNING = 5000,
  ERROR = 6000,
}

interface ToastData {
  severity: ToastSeverity;
  icon: ToastIcon;
  text: string;
  delay: number
}

export default class Toast implements ToastData {
  id: string;
  timestamp: number;
  severity: ToastSeverity;
  icon: ToastIcon;
  text: string;
  delay: number

  constructor ({ severity, icon, text, delay }: ToastData) {
    this.id = uuid.v4()
    this.timestamp = Date.now()
    this.severity = severity
    this.icon = icon
    this.text = text
    this.delay = delay
  }
}

export class SuccessToast extends Toast {
  constructor (text: string) {
    super({
      text,
      severity: ToastSeverity.SUCCESS,
      icon: ToastIcon.SUCCESS,
      delay: ToastDelay.SUCCESS
    })
  }
}

export class InfoToast extends Toast {
  constructor (text: string) {
    super({
      text,
      severity: ToastSeverity.INFO,
      icon: ToastIcon.INFO,
      delay: ToastDelay.INFO
    })
  }
}

export class WarningToast extends Toast {
  constructor (text: string) {
    super({
      text,
      severity: ToastSeverity.WARNING,
      icon: ToastIcon.WARNING,
      delay: ToastDelay.WARNING
    })
  }
}

export class ErrorToast extends Toast {
  constructor (text: string) {
    super({
      text,
      severity: ToastSeverity.ERROR,
      icon: ToastIcon.ERROR,
      delay: ToastDelay.ERROR
    })
  }
}
