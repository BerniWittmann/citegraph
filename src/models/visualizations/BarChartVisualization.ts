import Visualization, { VisualizationBaseFields } from './Visualization'

export interface BarChartVisualizationFields {}

export default class BarChartVisualization extends Visualization implements BarChartVisualizationFields {
  static key: string = 'barchart'
  key: string = 'barchart'
  static title: string = 'visualizations.information.bar_chart.title'
  static description: string = 'visualizations.information.bar_chart.description'
  static longDescription: string = 'visualizations.information.bar_chart.long_description'
  static imageUrl: string = 'visualization_images/bar_chart.png'

  constructor ({ id, name, progress }: VisualizationBaseFields & BarChartVisualizationFields) {
    super({ id, name, progress })
  }
}
