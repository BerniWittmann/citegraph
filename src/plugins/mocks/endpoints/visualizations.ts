import { AxiosRequestConfig } from 'axios'
import MockAdapter from 'axios-mock-adapter'
import db from '../db'
import Project from '@/models/project'

const projects = db.getCollection('projects')
const visualizations = db.getCollection('visualizations')

function getProjectId (config: AxiosRequestConfig): number {
  const urlParts: Array<string> = config.url!.split('/')
  return parseInt(urlParts[4])
}

export function setupMocks (mock: MockAdapter) {
  mock.onGet(/\/projects\/\d+\/visualizations$/).reply((config: AxiosRequestConfig) => {
    const project: Resultset<Project> = projects.findOne({ 'id': getProjectId(config) })
    if (!project) {
      return [404]
    }
    return [200, visualizations.find()]
  })
}
