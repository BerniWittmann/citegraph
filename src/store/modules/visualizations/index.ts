import { Module } from 'vuex'
import { VisualizationsState } from './types'
import { RootState } from '@/store/types'
import { state } from './state'
import { actions } from './actions'
import { getters } from './getters'
import { mutations } from './mutations'

const namespaced: boolean = true

export const visualizations: Module<VisualizationsState, RootState> = {
  namespaced,
  state,
  getters,
  actions,
  mutations
}
