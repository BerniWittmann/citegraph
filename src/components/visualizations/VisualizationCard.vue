<template>
  <v-card :hover="true"
          class="visualization-card"
          height="100%">
    <div class="d-flex flex-no-wrap justify-space-between">
      <v-avatar
        class="ma-3"
        size="125"
        tile
      >
        <v-img contain :src="imageUrl"/>
      </v-avatar>
      <div class="flex-grow-1 d-flex flex-column align-center">
        <v-card-title>{{ visualization.name }}</v-card-title>
        <v-card-text class="progress-information text-center">
          <span v-if="isLoading">{{ $t('visualizations.card.processing', { percent: progress }) }}</span>
          <span v-else>{{ $t('visualizations.card.ready') }}</span>
        </v-card-text>
        <v-divider style="width: 90%" class="mx-auto"></v-divider>
        <v-card-actions>
          <v-btn color="primary" text>
            {{ $t('visualizations.card.show') }}
          </v-btn>
          <v-btn color="primary" text>
            {{ $t('visualizations.card.edit') }}
          </v-btn>
        </v-card-actions>
      </div>
    </div>
    <v-progress-linear color="#F37F21" background-color="#FBBF35" background-opacity="0.5" v-show="isLoading" class="progress" :value="progress" :max="100"></v-progress-linear>
  </v-card>
</template>
<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator'
import Visualization from '../../models/visualizations/Visualization'
import { visualizationsKeyMap } from '@/models/visualizations'

@Component
export default class VisualizationCard extends Vue {
  @Prop() readonly visualization!: Visualization

  get visualizationClass () {
    return visualizationsKeyMap[this.visualization.key]
  }

  get imageUrl (): string {
    return `${process.env.BASE_URL}${this.visualizationClass.imageUrl}`
  }

  get progress (): number {
    return this.visualization.progress ? this.visualization.progress * 100 : 0
  }

  get isLoading (): boolean {
    return !!this.visualization.progress && (this.progress < 100)
  }
}
</script>

<style scoped lang="scss">
.progress-information {
  color: #F37F21;
}
</style>
