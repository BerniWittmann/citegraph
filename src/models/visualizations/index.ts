import WordCloudVisualization, { WordCloudVisualizationFields } from './WordCloudVisualization'
import { VisualizationBaseFields } from '@/models/visualizations/Visualization'
import BarChartVisualization, { BarChartVisualizationFields } from '@/models/visualizations/BarChartVisualization'

export const visualizations = [WordCloudVisualization, BarChartVisualization]

const visualizationsKeyMap: Record<string, any> = {}
visualizations.forEach((visualization) => {
  visualizationsKeyMap[visualization.key] = visualization
})

export {
  visualizationsKeyMap
}

export type VisalizationFields = VisualizationBaseFields & (
  WordCloudVisualizationFields &
  BarChartVisualizationFields
)
