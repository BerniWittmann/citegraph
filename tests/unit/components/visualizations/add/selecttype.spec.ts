import { shallowMount } from '@vue/test-utils'
import { i18n } from '../../../setupPlugins'

import VisualizationAddSelectTypeComponent from '@/components/visualizations/add/SelectType.vue'
import { visualizations } from '@/models/visualizations'

jest.useFakeTimers()

describe('components/visualizations/add/SelectType.vue', () => {
  const router = {
    back: jest.fn()
  }
  const getWrapper = () => {
    return shallowMount(VisualizationAddSelectTypeComponent, {
      i18n,
      mocks: {
        $router: router
      }
    })
  }

  it('renders', () => {
    const wrapper = getWrapper()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('has a back Button', () => {
    const wrapper = getWrapper()
    const btns = wrapper.findAll('v-btn-stub')
    const btn = btns.at(btns.length - 1)
    expect(btn.exists()).toBeTruthy()
    expect(btn.text()).toContain('visualizations.add.type.back')
    btn.vm.$emit('click')
    expect(router.back).toHaveBeenCalled()
  })

  visualizations.forEach((VisualizationClass: any, index: number) => {
    describe('renders a card for type ' + VisualizationClass.key, () => {
      let card: any
      let wrapper: any
      beforeEach(() => {
        wrapper = getWrapper()
        card = wrapper.findAll('v-col-stub').at(index).find('v-card-stub')
        expect(card.exists()).toBeTruthy()
      })
      it('renders', () => {
        expect(card.html()).toMatchSnapshot()
      })
      it('renders the title and description', () => {
        expect(card.text()).toContain(VisualizationClass.title)
        expect(card.text()).toContain(VisualizationClass.description)
        expect(card.text()).toContain(VisualizationClass.longDescription)
      })

      it('renders the image and description', () => {
        const img = card.find('v-img-stub')
        expect(img.exists()).toBeTruthy()
        expect(img.props('src')).toContain(VisualizationClass.imageUrl)
      })

      it('has a button to choose the type', () => {
        const btn = card.find('v-btn-stub')
        expect(btn.exists()).toBeTruthy()
        expect(btn.text()).toEqual('visualizations.add.type.choose')
        expect(wrapper.emitted('next-step')).toBeFalsy()
        expect(wrapper.emitted('update-type')).toBeFalsy()
        btn.vm.$emit('click')
        expect(wrapper.emitted('update-type')).toBeTruthy()
        expect(wrapper.emitted('update-type')[0][0]).toEqual(VisualizationClass)
        jest.runAllTimers()
        expect(wrapper.emitted('next-step')).toBeTruthy()
      })

      it('renders the card active if the type is active', () => {
        wrapper.setProps({
          currentType: VisualizationClass.key
        })
        expect(card.html()).toMatchSnapshot()
        expect(card.props('raised')).toBeTruthy()
      })
    })
  })
})
