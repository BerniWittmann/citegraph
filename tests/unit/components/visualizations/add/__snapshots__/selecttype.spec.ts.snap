// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/visualizations/add/SelectType.vue renders 1`] = `
<div class="visualization-add-type">
  <h1>visualizations.add.type.title</h1>
  <v-container-stub tag="div">
    <v-row-stub tag="div">
      <v-col-stub cols="12" md="4" tag="div">
        <v-card-stub loaderheight="4" tag="div" height="100%" hover="true">
          <div class="d-flex flex-no-wrap justify-space-between">
            <v-avatar-stub size="125" tile="true" class="ma-2">
              <v-img-stub contain="true" options="[object Object]" position="center center" src="visualization_images/word_cloud.jpg" transition="fade-transition"></v-img-stub>
            </v-avatar-stub>
            <div class="flex-grow-1 d-flex flex-column align-stretch">
              <v-card-title-stub>visualizations.information.word_cloud.title</v-card-title-stub>
              <v-card-text-stub>
                visualizations.information.word_cloud.description
              </v-card-text-stub>
              <v-divider-stub class="mx-auto" style="width: 90%;"></v-divider-stub>
              <v-card-actions-stub class="d-flex justify-space-around">
                <v-btn-stub color="primary" tag="button" activeclass="" text="true" type="button">
                  visualizations.add.type.choose
                </v-btn-stub>
                <v-menu-stub opendelay="0" closedelay="0" contentclass="" left="true" maxwidth="auto" nudgebottom="0" nudgeleft="0" nudgeright="0" nudgetop="0" nudgewidth="0" openonclick="true" closeonclick="true" closeoncontentclick="true" maxheight="auto" origin="top left" transition="v-menu-transition">
                  <v-card-stub loaderheight="4" tag="div" maxwidth="500px">
                    <v-card-title-stub class="d-flex justify-space-between">
                      visualizations.information.word_cloud.title
                      <v-btn-stub tag="button" activeclass="" icon="true" type="button">
                        <v-icon-stub>mdi-close</v-icon-stub>
                      </v-btn-stub>
                    </v-card-title-stub>
                    <v-card-text-stub class="d-flex justify-space-between align-center">
                      <v-avatar-stub size="175" tile="true" class="ma-3">
                        <v-img-stub contain="true" options="[object Object]" position="center center" src="visualization_images/word_cloud.jpg" transition="fade-transition"></v-img-stub>
                      </v-avatar-stub>
                      visualizations.information.word_cloud.long_description
                    </v-card-text-stub>
                  </v-card-stub>
                </v-menu-stub>
              </v-card-actions-stub>
            </div>
          </div>
        </v-card-stub>
      </v-col-stub>
      <v-col-stub cols="12" md="4" tag="div">
        <v-card-stub loaderheight="4" tag="div" height="100%" hover="true">
          <div class="d-flex flex-no-wrap justify-space-between">
            <v-avatar-stub size="125" tile="true" class="ma-2">
              <v-img-stub contain="true" options="[object Object]" position="center center" src="visualization_images/bar_chart.png" transition="fade-transition"></v-img-stub>
            </v-avatar-stub>
            <div class="flex-grow-1 d-flex flex-column align-stretch">
              <v-card-title-stub>visualizations.information.bar_chart.title</v-card-title-stub>
              <v-card-text-stub>
                visualizations.information.bar_chart.description
              </v-card-text-stub>
              <v-divider-stub class="mx-auto" style="width: 90%;"></v-divider-stub>
              <v-card-actions-stub class="d-flex justify-space-around">
                <v-btn-stub color="primary" tag="button" activeclass="" text="true" type="button">
                  visualizations.add.type.choose
                </v-btn-stub>
                <v-menu-stub opendelay="0" closedelay="0" contentclass="" left="true" maxwidth="auto" nudgebottom="0" nudgeleft="0" nudgeright="0" nudgetop="0" nudgewidth="0" openonclick="true" closeonclick="true" closeoncontentclick="true" maxheight="auto" origin="top left" transition="v-menu-transition">
                  <v-card-stub loaderheight="4" tag="div" maxwidth="500px">
                    <v-card-title-stub class="d-flex justify-space-between">
                      visualizations.information.bar_chart.title
                      <v-btn-stub tag="button" activeclass="" icon="true" type="button">
                        <v-icon-stub>mdi-close</v-icon-stub>
                      </v-btn-stub>
                    </v-card-title-stub>
                    <v-card-text-stub class="d-flex justify-space-between align-center">
                      <v-avatar-stub size="175" tile="true" class="ma-3">
                        <v-img-stub contain="true" options="[object Object]" position="center center" src="visualization_images/bar_chart.png" transition="fade-transition"></v-img-stub>
                      </v-avatar-stub>
                      visualizations.information.bar_chart.long_description
                    </v-card-text-stub>
                  </v-card-stub>
                </v-menu-stub>
              </v-card-actions-stub>
            </div>
          </div>
        </v-card-stub>
      </v-col-stub>
    </v-row-stub>
  </v-container-stub>
  <div class="my-10 d-flex">
    <v-btn-stub color="warning" tag="button" activeclass="" large="true" rounded="true" type="button">
      <v-icon-stub>mdi-menu-left</v-icon-stub> visualizations.add.type.back
    </v-btn-stub>
  </div>
</div>
`;

exports[`components/visualizations/add/SelectType.vue renders a card for type barchart renders 1`] = `
<v-card-stub loaderheight="4" tag="div" height="100%" hover="true">
  <div class="d-flex flex-no-wrap justify-space-between">
    <v-avatar-stub size="125" tile="true" class="ma-2">
      <v-img-stub contain="true" options="[object Object]" position="center center" src="visualization_images/bar_chart.png" transition="fade-transition"></v-img-stub>
    </v-avatar-stub>
    <div class="flex-grow-1 d-flex flex-column align-stretch">
      <v-card-title-stub>visualizations.information.bar_chart.title</v-card-title-stub>
      <v-card-text-stub>
        visualizations.information.bar_chart.description
      </v-card-text-stub>
      <v-divider-stub class="mx-auto" style="width: 90%;"></v-divider-stub>
      <v-card-actions-stub class="d-flex justify-space-around">
        <v-btn-stub color="primary" tag="button" activeclass="" text="true" type="button">
          visualizations.add.type.choose
        </v-btn-stub>
        <v-menu-stub opendelay="0" closedelay="0" contentclass="" left="true" maxwidth="auto" nudgebottom="0" nudgeleft="0" nudgeright="0" nudgetop="0" nudgewidth="0" openonclick="true" closeonclick="true" closeoncontentclick="true" maxheight="auto" origin="top left" transition="v-menu-transition">
          <v-card-stub loaderheight="4" tag="div" maxwidth="500px">
            <v-card-title-stub class="d-flex justify-space-between">
              visualizations.information.bar_chart.title
              <v-btn-stub tag="button" activeclass="" icon="true" type="button">
                <v-icon-stub>mdi-close</v-icon-stub>
              </v-btn-stub>
            </v-card-title-stub>
            <v-card-text-stub class="d-flex justify-space-between align-center">
              <v-avatar-stub size="175" tile="true" class="ma-3">
                <v-img-stub contain="true" options="[object Object]" position="center center" src="visualization_images/bar_chart.png" transition="fade-transition"></v-img-stub>
              </v-avatar-stub>
              visualizations.information.bar_chart.long_description
            </v-card-text-stub>
          </v-card-stub>
        </v-menu-stub>
      </v-card-actions-stub>
    </div>
  </div>
</v-card-stub>
`;

exports[`components/visualizations/add/SelectType.vue renders a card for type wordcloud renders 1`] = `
<v-card-stub loaderheight="4" tag="div" height="100%" hover="true">
  <div class="d-flex flex-no-wrap justify-space-between">
    <v-avatar-stub size="125" tile="true" class="ma-2">
      <v-img-stub contain="true" options="[object Object]" position="center center" src="visualization_images/word_cloud.jpg" transition="fade-transition"></v-img-stub>
    </v-avatar-stub>
    <div class="flex-grow-1 d-flex flex-column align-stretch">
      <v-card-title-stub>visualizations.information.word_cloud.title</v-card-title-stub>
      <v-card-text-stub>
        visualizations.information.word_cloud.description
      </v-card-text-stub>
      <v-divider-stub class="mx-auto" style="width: 90%;"></v-divider-stub>
      <v-card-actions-stub class="d-flex justify-space-around">
        <v-btn-stub color="primary" tag="button" activeclass="" text="true" type="button">
          visualizations.add.type.choose
        </v-btn-stub>
        <v-menu-stub opendelay="0" closedelay="0" contentclass="" left="true" maxwidth="auto" nudgebottom="0" nudgeleft="0" nudgeright="0" nudgetop="0" nudgewidth="0" openonclick="true" closeonclick="true" closeoncontentclick="true" maxheight="auto" origin="top left" transition="v-menu-transition">
          <v-card-stub loaderheight="4" tag="div" maxwidth="500px">
            <v-card-title-stub class="d-flex justify-space-between">
              visualizations.information.word_cloud.title
              <v-btn-stub tag="button" activeclass="" icon="true" type="button">
                <v-icon-stub>mdi-close</v-icon-stub>
              </v-btn-stub>
            </v-card-title-stub>
            <v-card-text-stub class="d-flex justify-space-between align-center">
              <v-avatar-stub size="175" tile="true" class="ma-3">
                <v-img-stub contain="true" options="[object Object]" position="center center" src="visualization_images/word_cloud.jpg" transition="fade-transition"></v-img-stub>
              </v-avatar-stub>
              visualizations.information.word_cloud.long_description
            </v-card-text-stub>
          </v-card-stub>
        </v-menu-stub>
      </v-card-actions-stub>
    </div>
  </div>
</v-card-stub>
`;
